app/api/attendees.py

```diff
@@ -149,6 +149,18 @@ def before_update_object(self, obj, data, kwargs):
                 if obj.checkin_times and data['checkin_times'] not in obj.checkin_times.split(","):
                     data['checkin_times'] = '{},{}'.format(obj.checkin_times, data['checkin_times'])
 
+        if 'is_checked_out' in data and data['is_checked_out']:
+            if 'checkout_times' not in data:
+                raise UnprocessableEntity(
+                    {'pointer': '/data/attributes/checkout_times'},
+                    "Check out time missing while trying to check out"
+                    "attendee")
```

Comment:
This means we're expecting the client to send in the checkout_times. Why can't we just use the current time in the server for it ?

Reply:
So the column should be present in the model but this field should be absent from the attendee schema? Also, after what action should the current time be specified as the checkout_time in the server?

Reply:
sorry, but I didn't understand what you meant by the first question. 
All I am trying to say is that once a PATCH request is made to checkout the attendee, then do save the current time as one of the checkout times in the after_update_object method.

Reply:
Okay, I think I misinterpreted you - I thought you were saying that `checkout_times` should be a read-only value.

Reply:
We do something similar in columns such as `created_at` although it is a one time action so it is done inside the constructor. Here attendee can be checked out multiple times. So just check for it in the `before_update_object` and add the current time into the data as one of the checkout times.

**Note**

In my above comment, I accidently said `after_update_object`. Use `before_update_object`.

Reply:
Yes, makes sense. If we see the use case of the checkout button. It will be called whenever user checkout from the event so this we can definitely make use of current timestamps. And yes this way the `checkout_times` will be read only.